ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -689,10 +700,9 @@ int mca_pml_ucx_isend(const void *buf, size_t count, ompi_datatype_t *datatype,
                                  PML_UCX_MAKE_SEND_TAG(tag, comm));
     }
 
-    req = (ompi_request_t*)ucp_tag_send_nb(ep, buf, count,
-                                           mca_pml_ucx_get_datatype(datatype),
-                                           PML_UCX_MAKE_SEND_TAG(tag, comm),
-                                           mca_pml_ucx_send_completion);
+    req = mca_pml_ucx_tag_send_nb(ep, buf, count, mca_pml_ucx_get_datatype(datatype),
```

Comment:
currently for regular send the mode is checked twice (one to compare to BUFFERED, second time to compare to SYNCHRONOUS. Need to avoid the extra branch and reduce code duplication